microbenchmarks/singledispatch.py

```diff
@@ -0,0 +1,53 @@
+from functools import singledispatch
+
+from benchmarking import benchmark
+
+NUM_ITER = 100_000
```

Comment:
Right now this takes about 2m per iteration on my laptop. I've been trying to get microbenchmark runs to typically take less than a second when interpreted. Can you reduce the iteration to count to, say, 500? This should do the trick. The runner will run the benchmark several times, so a 2 minute iteration results in quite slow runtimes.

Reply:
Fixed in de1651891090ca5e3464e61477f02be76b10103d. The benchmark reports times of approximately 0.6 seconds now on my computer (as compared to over 7 minutes for the previous number of iterations), so the benchmark should be fast enough now.